zookeeper-server/src/main/java/org/apache/zookeeper/server/persistence/FileTxnSnapLog.java

```diff
@@ -499,23 +499,28 @@ public void save(
      * @return true if able to truncate the log, false if not
      * @throws IOException
      */
-    public boolean truncateLog(long zxid) throws IOException {
-        // close the existing txnLog and snapLog
-        close();
-
-        // truncate it
-        FileTxnLog truncLog = new FileTxnLog(dataDir);
-        boolean truncated = truncLog.truncate(zxid);
-        truncLog.close();
-
-        // re-open the txnLog and snapLog
-        // I'd rather just close/reopen this object itself, however that
-        // would have a big impact outside ZKDatabase as there are other
-        // objects holding a reference to this object.
-        txnLog = new FileTxnLog(dataDir);
-        snapLog = new FileSnap(snapDir);
-
-        return truncated;
+    public boolean truncateLog(long zxid) {
+        try {
+            // close the existing txnLog and snapLog
+            close();
+
+            // truncate it
+            FileTxnLog truncLog = new FileTxnLog(dataDir);
+            boolean truncated = truncLog.truncate(zxid);
+            truncLog.close();
```

Comment:
what about calling this **close** in a finally block ?

```
try { 
    FileTxnLog truncLog = new FileTxnLog(dataDir);
            boolean truncated = truncLog.truncate(zxid);
  } finally {
            truncLog.close();
}
```

Reply:
this is done slightly differently. please check.